primitives/src/opcodes.rs

```diff
@@ -71,6 +72,40 @@ macro_rules! all_opcodes {
                 }
             }
         }
+
+        impl str::FromStr for Opcode {
+            type Err = ();
+            fn from_str(s: &str) -> Result<Self, Self::Err> {
+                match s {
+                    "OP_0" => Ok(OP_0),
+                    "OP_TRUE" | "TRUE" => Ok(OP_TRUE),
+                    "OP_FALSE" | "FALSE" => Ok(OP_FALSE),
+                    "OP_NOP2" | "NOP2" => Ok(OP_NOP2),
+                    "OP_NOP3" | "NOP3" => Ok(OP_NOP3),
```

Comment:
This PR has code in in that we discussed in #3617 and decided not to use however the patch makes no attempt to justify _why_ this is now a valid thing to do. If you think that we were wrong in #3617 please add a PR that fully justifies _why_ this is a good change and makes a case for why the conclusion in #3617 was wrong.

Reply:
And FTR the orphan rule is not a valid reason because if a user wants an ad-hoc string parsing function they can write one (just not implement `FromStr for Opcode`) - we don't want to add a `FromStr` impl that suits one set of users and not another.